liberty-managed/src/main/java/io/openliberty/arquillian/managed/WLPManagedContainer.java

```diff
@@ -815,6 +799,43 @@ public void undeploy(final Archive<?> archive) throws DeploymentException
          log.exiting(className, "undeploy");
       }
    }
+ 
+   private void deleteWithRetries(File archveToDelete, boolean isFailSafeUndeployment, int retries, int retryInterval) throws DeploymentException {
+      int retryNo = -1;
+      boolean complete = false;
+      do {
+         if (retryNo >= 0) {
+            try {
+               Thread.sleep(retryInterval);
+            } catch (InterruptedException e) {
+               throw new DeploymentException("Interrupted Exception", e);
+            }
+         }
+         if (!containerConfiguration.isFailSafeUndeployment()) {
+            try {
+               if (!Files.deleteIfExists(archveToDelete.toPath())) {
+                  throw new DeploymentException("Archive already deleted from deployment directory");
```

Comment:
Now that we're doing retries, we probably want to reevaluate throwing exceptions here (and elsewhere in the code)

Reply:
I missed here that `deleteIfExists` returns `false` if the file doesn't exist but throws an exception if it does exist but can't be deleted.

I'm still a bit unsure as to whether we should throw a deployment exception here. I guess it's an unexpected case, but on the other hand the thing we wanted to get rid of is gone. I can't decide whether treating this as an error case is at odds with the idea of retrying the operation until the file is deleted.